ui/summary.go

```diff
@@ -79,15 +83,20 @@ func VerifyTrendColumnStat(stat string) error {
 }
 
 // UpdateTrendColumns updates the default trend columns with user defined ones
-func UpdateTrendColumns(stats []string) {
-	newTrendColumns := make([]TrendColumn, 0, len(stats))
+func UpdateTrendColumns(metricStats []string) {
+	newTrendColumns := make([]TrendColumn, 0, len(metricStats))
 
-	for _, stat := range stats {
-		percentileTrendColumn, err := generatePercentileTrendColumn(stat)
-
-		if err == nil {
-			newTrendColumns = append(newTrendColumns, TrendColumn{stat, percentileTrendColumn})
+	for _, stat := range metricStats {
```

Comment:
Using `for i, stat := range metricStats`, then assigning by `newTrandColumns[i] = ...` will be better, since when we did allocate the `newTrendColumns` already with the same cap with `metricStats`.

Reply:
I agree with this.

Reply:
I like this suggestion, but after trying it, it unfortunately breaks the [`TestUpdateTrendColumns/Ignore invalid stats`](https://github.com/loadimpact/k6/blob/ac255fc58de52e9c002e94fc5cfd73e3fb5be8b1/ui/summary_test.go#L111-L119) test.

Notice that we can't assume the length of `metricStats` upfront, since some of the column names might be invalid and should be skipped. So `newTrendColumns := make([]TrendColumn, len(metricStats))` allocates an array of size 3, but the expected size of generated columns should be 2 in this case.

I wouldn't want to resize or clean up the array afterwards, so I'd prefer it if we stuck to the original `append()` implementation.

What do you think?

Reply:
I don't think we need to over-optimize a piece of code that would be called once in a test run, and what's more, after the end of the actual test run, when a microscopic performance improvement doesn't matter one jot.